metrotranscom/doorway
# fix: add user and application import scripts

This PR addresses [#4193](https://github.com/bloom-housing/bloom/issues/4193)


## Description

Creates a script that does the following two things:
- Pulls over all partner users in HBA tied to at least one listing in the requested jurisdiction. Then ties it to the same listing in Doorway
- Pulls over all public users in HBA that have at least one application submitted in the requested jurisdiction. Then copies over all of those applications to Doorway
  - No PII is carried over for the applications

## How Can This Be Tested/Reviewed?

This requires the previous jurisdiction scripts to be run so the following steps need to be run:
- Do a reseed locally
- Start up the app locally and go to the swagger api http://localhost:3100/api.
- Login via the api to the admin account http://localhost:3100/api#/auth/login.
- Go to Heroku and get the DB credentials for the HBA staging environment https://data.heroku.com/datastores/fd968c98-f208-4066-b8cf-7d3a6bf8ff03#administration
- Use the credentials from above to run the first script http://localhost:3100/api#/scriptRunner/transferJurisdictionData.
- Run the script for http://localhost:3100/api#/scriptRunner/transferJurisdictionListingsData using the same db credentials and jurisdiction used in the previous script
- Run the latest script to migrate users and applications http://localhost:3100/api#/scriptRunner/transferJurisdictionUserApplicationData.
- Verify that the expected partner users from HBA staging have been carried over. You can verify partner users by logging into the partner site and going to the /users tab. Note: you need to be logged in as an admin to see those users. 
- Verify public users have been transferred over. The easiest way is to compare the user_accounts and associated tables in the database 
- Verify that applications have properly been carried over. You can do this by looking at the properties in the partner site and clicking on the applications tab and then the specific application you want to look at.

Notes:
- Testing with SMC will give you data, but there is much more data in HBA staging for Alameda. If you want to test with Alameda you will notice the following:
  - The Alameda jurisdiction isn't in the system so you will need to either manually add it or add it in the seeding script.
  - There will be a lot of logs output in the listing script because the data isn't great and also not every reserved type is carried over. This is ok, we only care about making sure the applications and users are transferred.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
